### ISSUE TITLE
Making part name tags a separate mod

### ISSUE THREAD

**Issue by djungelorm** [KSP-KOS/KOS#1742]
I had a brief chat with @Dunbaratu recently regarding moving the part name tags into a separate mod, so that kOS and kRPC scripts can share the same name tags. If we separate it into a separate mod, then configure it as a CKAN dependency for kOS and kRPC that should work well.

If you're still happy to do this, could we get the ball rolling? Thanks!

Original discussion here: http://forum.kerbalspaceprogram.com/index.php?/topic/130742-113-krpc-control-the-game-using-c-c-java-lua-python-ruby-haskell-v035-25th-june-2016/&page=5#comment-2668996

**Comment by hvacengi** [comment]
I suppose it should be rolled into it's own repository at that point.  It wouldn't even need to be a full on dependency of kOS at that point, just "recommended" (or whatever CKAN listing will try to install it by default).  I know that there was much discussion regarding maintaining a user's ability to remove the tags via MM originally, and completely removing the dll and configs might be even better in terms of performance.

Looks like it would mean breaking out `/src/kOS/Module/KOSNameTag.cs` as well as the window logic (`/src/kOS/Screen/KOSNameTagWindow.cs`).  Functionally, it may mean that kOS needs to use a reflection api to access the data, since we would no longer be guaranteed that the dll is installed at the same time.

**Comment by djungelorm** [comment]
I've extracted this functionality into a separate mod here: https://github.com/krpc/NameTag

I've kept the name of the part module as "KOSNameTag" to preserve backwards compatibility with old craft files. For example, if someone created a craft with kOS installed, then replaced kOS with NameTag, it would see the tag value set by kOS. This works in the other direction too.

I've added part tags support to kRPC (https://github.com/krpc/krpc/commit/38b2f161b07d655cfd9ae9aeb32036ed089272e9) - it just looks for a "KOSNameTag" part module using reflection. This allows people to either have kOS or NameTag installed to get the functionality.

I've only released it on github so far, but plan to release it on CKAN at some point, as a recommended dependency for kRPC. As having both kOS and NameTag installed would cause issues, I'm thinking we can set a conflict between the two mods so that CKAN won't install both at the same time.

Lastly, I think I've maintained the licenses correctly - please shout at me if I did something wrong! Thanks!

**Comment by hvacengi** [comment]
I'll poke in to the project this weekend to review.  One question, if a user has both kOS and the NameTag mod installed, is there an issue?  If kOS will need to migrate to using the same mod in order to support the nametags, we'll want to coordinate the release (possibly for the next maintenance release).

**Comment by Dunbaratu** [comment]
I think even if kOS doesn't *have* to have the nametag stuff removed to use it from the separate mod instead, kOS probably *should* be changed to do so.  Just for code maintenence it's good not do duplicate code.

**Comment by hvacengi** [comment]
Absolutely.  It's just higher priority to get migrated sooner if this means that functionality is missing if both kRPC and kOS are installed.

**Comment by djungelorm** [comment]
I haven't tested it, but if both NameTag and kOS are installed then I think all that will happen is that the UI will get duplicated in the right click menu.

kOS doesn't *need* to migrate, but the user shouldn't have both the current version of kOS and NameTag installed at the same time, i.e. valid combinations with all the functionality working would be:

* kOS only
* kRPC + NameTag
* kRPC + kOS

**Comment by bzm3r** [comment]
Is there any interest in this?
